tls/s2n_resume.c

```diff
@@ -62,6 +65,29 @@ static int s2n_serialize_resumption_state(struct s2n_connection *conn, struct s2
     return 0;
 }
 
+static S2N_RESULT s2n_tls13_serialize_resumption_state(struct s2n_connection *conn, struct s2n_ticket_fields *ticket_fields,
+        struct s2n_stuffer *out)
+{
+    ENSURE_REF(conn);
```

Comment:
Do we want to ensure that the TLS version is 1.3 prior to serialization?

Reply:
Eh, we don't usually check the tls version when we call an s2n_tls13 function.

Reply:
We actually do that in some places and I think it is a valid check to include within the function or prior to function call to ensure the function is indeed called only in TLS1.3 versions. 

Ref: 

https://github.com/awslabs/s2n/blob/main/tls/s2n_client_finished.c#L60
https://github.com/awslabs/s2n/blob/main/tls/s2n_client_finished.c#L89
https://github.com/awslabs/s2n/blob/main/tls/s2n_server_finished.c#L79
https://github.com/awslabs/s2n/blob/main/tls/s2n_server_finished.c#L110

Reply:
It's maybe redundant to check the protocol version, call a method with tls13 in its name, then check the protocol version again.

Reply:
To clarify, my suggestion was not to include both checks, but either - prior to function call OR within the function. Currently we do the check prior to function and satisfies the requirement.